docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,535 @@
+
+```
+GET /wp/v2/widget-areas/footer
+```
+
+#### Response
+
+- Status:
+  - `200` if successful
+  - `404` if requested widget area does not exist
+- Body: A single widget area resource
+
+```json
+{
+	"id": "footer",
+	"name": "Footer",
+	"description": "",
+	"content": {
+		"raw": "<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->",
+		"rendered": "<p>Hello there!</p>"
+	},
+}
+```
+
+### Updating a single widget area
+
+Updates the blocks in a single registered widget area.
+
+#### Request
+
+- Method: `PUT`
+- Route: `/wp/v2/widget-areas/:id`
+- Parameters:
+  - `id`: The unique identifier of the widget area to update
+- Body: A single widget area resource
+
+```
+PUT /wp/v2/widget-areas/footer
+
+{ "content": { "raw": "<!-- wp:block {\"ref\":123} /-->" } }
+```
+
+#### Response
+
+- Status:
+  - `200` if successful
+  - `404` if specified widget area does not exist
+  - `400` if request body is not a valid widget area resource
+- Body: The updated widget area resource
+
+```json5
+{
+	"id": "footer",
+	"name": "Footer",
+	"description": "",
+	"content": {
+		"raw": "<!-- wp:block {\"ref\":123} /-->",
+		"rendered": "<p>This is a reusable paragraph block</p>"
+	},
+}
+```
+
+## Storage
+
+WordPress currently stores which widgets belong to which widget areas using the `'sidebars_widgets'` site option. This is a serialized PHP array that maps widget area identifiers to a list of widget identifiers that are in that area.
+
+```php
+array(
+	'footer' => array(
+		'search-2',
+		'recent-posts-2',
+		'recent-comments-2',
+	),
+	'sidebar' => array(
+		'recent-posts-2',
+		'recent-comments-2',
+	),
+	'array_version' => 3,
+)
+```
+
+When a widget area stored this way is accessed via the REST API, widgets will be represented using `core/legacy-widget` blocks. Each `core/legacy-widget` block stores a reference to the existing widget (`identifier`), and, if it is a multi widget, the attributes that that instance has (`instance`).
+
+```
+GET /wp/v2/widget-areas/footer
+```
+
+```json
+{
+	"id": "footer",
+	...
+	"content": {
+		"raw": "
+			<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-posts-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-comments-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+		",
+		"rendered": ...,
+	}
+}
+```
+
+When a widget area is updated using the REST API, the storage format is _migrated_ and its `'array_version'` is set to `4`.
+
+After migration, the widget area identifier no longer maps to an array of widget identifiers. Instead, it maps to the ID of a `wp_area` post.
+
+`wp_area` is a new post type that denotes posts that contain block markup for a widget area. The `wp_area` post type is similar to the `wp_block` post type, except that it is marked as private and therefore not viewable by accessing `wp-admin/edit.php?post_type=wp_area`.
```

Comment:
Will this new post type support revisions? How about authors?

Reply:
Interesting! revisions could possible be good to have but I guess we should aim for the minimal solution first and iterate on enhancements.

Reply:
We could leave them turned on, with new ui and then plugin developers could add this as a feature project.

Reply:
Currently supporting previews in the customizer where it shows new versions of widget instances with changes but does not really saves them until a publish is really complex. Having the concept of published block area and draft block area may be very useful. During the previews, we use the last drafts of the block areas while still using the published version on the normal view.

Reply:
I think for now, no, but part of what persuaded me to use `wp_area` posts is that it steers us towards a future where all sections of your site are version controlled and support publishing flows.

Reply:
This is how the customizer changesets were created. They use post type, becuases you have the ability to draft and send previews to others using a special link because revisions. If we don't turn them on from the start, we will use that data forever.